config/configtls/configtls.go

```diff
@@ -135,7 +143,7 @@ func (r *certReloader) GetCertificate() (*tls.Certificate, error) {
 	// If a reload is in progress this will block and we will skip reloading in the current
 	// call once we can continue
 	r.lock.RLock()
-	if r.ReloadInterval != 0 && r.nextReload.Before(now) {
+	if r.ReloadInterval != 0 && r.nextReload.Before(now) && r.CertFile != "" {
```

Comment:
I think this should probably handle mixed inline and file-based TLS settings for  the CA, cert, and key.